### ISSUE TITLE
Multiselect: paste value

### ISSUE THREAD

**Issue by adavidof** [wix/wix-style-react#2559]
# Relevant Components
#### Multiselect
`import WixMultiselect from 'wix-style-react/MultiSelect';`

Paste text(Cmd + V)

### Current Behavior
Paste to input some text
The `onChange` callback is fired twice with empty value
The `onSelect` callback is fired

### Expected Behavior
Paste to input some text
The `onManuallyInput` callback once is fired with proper value

## Steps to Reproduce
https://codesandbox.io/s/xv0n1zppop

## Severity

Blocker

**Comment by adavidof** [comment]
Regarding https://github.com/wix/wix-style-react/issues/2199 as well

**Comment by shlomitc** [comment]
Just for the reference - the Tags/MultiSelect component is a bit complex regarding what should happen on a new value. it acts both as a "Select" (pick from list) component and as an "Input" (add new items).

The behavior you're describing in the second issue is relevant to the "Input" family, and this is currently achieved with the `onManuallyInput` prop.

I think we can achieve this also with `onBlur` but for now I think you can use `onManuallyInput`

**Comment by adavidof** [comment]
@adavidof yeah, but in the example https://codesandbox.io/s/xv0n1zppop `onManuallyInput` doesn't work :/

**Comment by shlomitc** [comment]
Try the following snippet, it's the most "controlled" way you can work with the component.
```jsx
import React from 'react';
import MultiSelect from 'wix-style-react/MultiSelect';

class App extends React.Component {
  state = {
    inputValue: '',
    tags: [],
  };

  uniqueTagId = 0;

  handleOnSelect = tags => {
    const newTag = { ...tags[0], id: this.uniqueTagId.toString() };
    this.uniqueTagId++;
    return this.setState({ tags: [...this.state.tags, newTag] });
  };

  handleOnRemoveTag = tagId =>
    this.setState({
      tags: this.state.tags.filter(currTag => currTag.id !== tagId),
    });

  handleOnChange = event => this.setState({ inputValue: event.target.value });

  render() {
    return (
      <MultiSelect
        // control the input changes
        value={this.state.inputValue}
        onChange={this.handleOnChange}
        // the actual tags to display
        tags={this.state.tags}
        // manage tags creation and removal
        onSelect={this.handleOnSelect}
        onRemoveTag={this.handleOnRemoveTag}
      />
    );
  }
}

export default App;
```

**Comment by adavidof** [comment]
@shlomitc 
I changed priority to blocker: these bugs are blocking us.

**Comment by hepiyellow** [comment]
## New API suggestion

I think having 2 callback `onSelect` and `onManuallyInput`  is not appropriate for this component.

- A single `onTagsAdded` should be enough.
- The consumer shouldn't care how the tag was created (from a suggestion, or by manually typing it)
- The consumer can always check if the new tag was in the options or not.

I am going to check this option, and see if "it holds water".

**Comment by hepiyellow** [comment]
@adavidof 


| Options | Paste | Type & Enter | Calls |
|---|---|---|---|
| without | new | |onSelect(customId, new) |
| without |  | new |onSelect, onManuallyInput |
| with | alabama|  |onSelect |
| with | alab|  |onSelect(customId,alab) |
| with | new | |onSelect(customId, new) |
| with | | alabama |onSelect, onManuallyInput |
| with | | alab |onSelect, onManuallyInput |
| with | | new | onManuallyInput(new) |

- `Options` -  "with" it means there is are options, and "alabama" is one of them.
- `new` means that its a value which is not in the options
- `onSelect` - sometimes`onSelect

**Comment by hepiyellow** [comment]
## UX questions

### Tag with `theme='error'`

<img width="746" alt="screen shot 2018-12-25 at 19 37 12" src="https://user-images.githubusercontent.com/6338722/50425712-8ca37400-087c-11e9-8fbd-f7d046709eaa.png">

Only when pasting a new tag (that doesn't exist in options) we call `onSelect` with a new tag that has `theme='error'`. (It doesn't happen if a new tag is manually submitted)

- What is the logic behind this?
- The Zeplin specs don't say anything about when to create an `error` tag
- What is the correct behaviour for new tag?
- Isn't it the decission of the consumer if to mark a tag as an `error` tag?

Need to ask UX  @wuwa @milkyfruit @LittleTal

**Comment by hepiyellow** [comment]
## Approach 3 - Paste should not submit

Same as [Vue-Multiselect](https://vue-multiselect.js.org/#sub-tagging), pasting text would not submit anything. So it's up to the user to either select from options or press enter to add a new tag.
Back to the usual flow.

Certainly, NOT a breaking change.
Just different user experience.

@shlomitc @adavidof ?

**Comment by wuwa** [comment]
my thoughts:
1. error tag might be used in different situations:
- not in predefined list (but still consumer wants to show it)
- not valid (for example email format)
- not validated async (for example pasting many emails and checking if they are in some group)
2. so from the options 3 is not good. 1 or 2 - not sure i fully follow everything but 1 sounds better to me.
3. to sum up - i believe that the logic should be split - entering tags is always the same (doesn't matter if by typing, selecting, or pasting. then each entry should be either "added", "discarded" or "error", depends on the setup of the component.

**Comment by hepiyellow** [comment]
## Remove Auto-Tag-Creation-On-Paste

@adavidof We are going to remove the special handling of Paste.

When pasting is done, we won't call nither `onSelect` nor `onManuallyInput`.
After pasting, the user needs to press the "Enter" key or select from the options.

**Comment by hepiyellow** [comment]
Resolved in Release 5.24.0
